sfepy/discrete/fem/lcbc_operators.py

```diff
@@ -141,6 +141,72 @@ def __init__(self, name, regions, dof_names, dof_map_fun,
         self.n_new_dof = n_rigid_dof
         self.mtx = self.mtx[indx]
 
+class Rigid2Operator(LCBCOperator):
+    """
+    Transformation matrix operator for rigid body multi-point constraint LCBCs.
+    """
+    kind = 'rigid2'
+
+    def __init__(self, name, regions, dof_names, dof_map_fun,
+                 variables, ts=None, functions=None):
+        LCBCOperator.__init__(self, name, regions, dof_names, dof_map_fun,
+                              variables, functions=functions)
+
+        mvar, svar = variables[self.var_names[0]], variables[self.var_names[1]]
+        mfield, sfield = mvar.field, svar.field
+
+        mnodes = mfield.get_dofs_in_region(regions[0], merge=True)
+        snodes = sfield.get_dofs_in_region(regions[1], merge=True)
+        if len(snodes) != 1:
+            raise ValueError('Number of independent nodes must be 1! (is %d)'
+                             % len(snodes))
+
+        self.mdofs = expand_nodes_to_equations(mnodes, dof_names[0],
+                                               self.all_dof_names[0])
+        self.sdofs = expand_nodes_to_equations(snodes, dof_names[1],
+                                               self.all_dof_names[1])
+
+        meq, seq = mvar.eq_map.eq[self.mdofs], svar.eq_map.eq[self.sdofs]
+
+        assert_(nm.all(meq >= 0))
+        #assert_(nm.all(seq >= 0))
+
+        mcoors = mfield.get_coor(mnodes)
+        scoors = sfield.get_coor(snodes)
+
+        coors = mcoors - scoors
+        n_nod, dim = coors.shape
+
+        mtx_e = nm.tile(nm.eye(dim, dtype=nm.float64), (n_nod, 1))
+
+        if dim == 2:
+            mtx_r = nm.empty((dim * n_nod, 1), dtype=nm.float64)
+            mtx_r[0::dim,0] = -coors[:,1]
+            mtx_r[1::dim,0] = coors[:,0]
+
+        elif dim == 3:
+            mtx_r = nm.zeros((dim * n_nod, dim), dtype=nm.float64)
+            mtx_r[0::dim,1] = coors[:,2]
+            mtx_r[0::dim,2] = -coors[:,1]
+            mtx_r[1::dim,0] = -coors[:,2]
+            mtx_r[1::dim,2] = coors[:,0]
+            mtx_r[2::dim,0] = coors[:,1]
+            mtx_r[2::dim,1] = -coors[:,0]
+
+        else:
+            msg = 'dimension in [2, 3]: %d' % dim
```

Comment:
Perhaps a better error message?

`Expected dimension 2 or 3 but got ...`

Reply:
Thanks, fixed. Note that it is still a work in progress, so further changes are to be expected.